infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutSwitchController.kt

```diff
@@ -31,41 +31,30 @@ class LayoutSwitchController(
     fun getTrackLayoutSwitches(
         @PathVariable("publishType") publishType: PublishType,
         @RequestParam("bbox") bbox: BoundingBox?,
-        @RequestParam("name") name: String?,
+        @RequestParam("namePart") namePart: String?,
+        @RequestParam("exactName") exactName: SwitchName?,
         @RequestParam("offset") offset: Int?,
         @RequestParam("limit") limit: Int?,
         @RequestParam("comparisonPoint") comparisonPoint: Point?,
         @RequestParam("switchType") switchType: String?,
         @RequestParam("includeSwitchesWithNoJoints") includeSwitchesWithNoJoints: Boolean = false,
+        @RequestParam("includeDeleted") includeDeleted: Boolean = false,
     ): List<TrackLayoutSwitch> {
         logger.apiCall(
             "getTrackLayoutSwitches",
             "publishType" to publishType,
             "bbox" to bbox,
-            "name" to name,
+            "namePart" to namePart,
+            "exactName" to exactName,
             "offset" to offset,
             "limit" to limit,
             "comparisonPoint" to comparisonPoint,
-            "switchType" to switchType
-        )
-        val filter = switchService.switchFilter(name, switchType, bbox, includeSwitchesWithNoJoints)
-        return switchService.pageSwitchesByFilter(publishType, filter, offset, limit, comparisonPoint)
-    }
-
-    @PreAuthorize(AUTH_ALL_READ)
-    @GetMapping("/{publicationState}/by-name")
```

Comment:
Review-pointti 1: yhdistetään ylläolevaan kun kuitenkin on kyse aika samasta asiasta ja näin tulee fiksumman näköinen API. Iso parametrimäärä kasvaa vähän entisestään, mutta tämä on efektiivisesti haku-api josta voi hakea haluamallaan combolla suodatusehtoja, joten en näe tuota ihan niin pahana. Jos joku tietää paremman tavan tehdä restimäisesti tuo niin voidaan toki rempata. Melko yleinen on POST jossa bodyssä tulee searchParams, mutta itse jotenkin tykkään getistä monella queryparamilla enemmän, sillä se on semanttisesti oikeampi.

Reply:
Tämä on mielestäni ihan jees näin. Mitä noihin parametrimääriin tulee niin joo, onhan tämä tosiaan aika hirvitys, mutta en oikein nää että näille voisi kauheasti mitään. Toisaalta jos joku funktio tarvitsee paljon parametreja (niinkuin tämä esimerkiksi tarvii), niin turha niitä sinällään mihinkään Params-tietotyyppiinkään on kapseloida. Ihan samalla tavalla ne pitäisi silti asettaa sinne, ja pohjaongelma pysyisi ihan samana (korkeintaan ehkä hitusen piilotettuna)